src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3485,6 +3486,44 @@ internal void SymbolDeclaredEvent(Symbol symbol)
             EventQueue?.TryEnqueue(new SymbolDeclaredCompilationEvent(this, symbol.GetPublicSymbol()));
         }
 
+        internal override void SerializePdbEmbeddedCompilationOptions(BlobBuilder builder)
+        {
+            WriteValue("checked", Options.CheckOverflow.ToString());
+            WriteValue("nullable", Options.NullableContextOptions.ToString());
+            WriteValue("unsafe", Options.AllowUnsafe.ToString());
+            WriteValue("langversion", LanguageVersion.ToDisplayString());
+
+            var isOptimized = Options.OptimizationLevel == OptimizationLevel.Release ? true : false;
+            WriteValue("optimize", isOptimized.ToString());
```

Comment:
Why write this as a `bool` when the other `enum` options are written as `string`?

Reply:
This is stored as an enum internally, but is a bool commandline value using +/-

Reply:
There are 2 command line switches that influence the enum: `/debug+`, `/optimize(+|-)`.

---
In reply to: [434096192](https://github.com/dotnet/roslyn/pull/44373#discussion_r434096192) [](ancestors = 434096192)

Reply:
We need to also include debug+ mode: [http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/CommandLine/CSharpCommandLineParser.cs,1425](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/CommandLine/CSharpCommandLineParser.cs,1425)

I'm not sure why is it not a third value in OptimizationLevel, but we should encode it as such. So this should be `optimize` -> (`release`|`debug`|`debug+`)

---
In reply to: [434220679](https://github.com/dotnet/roslyn/pull/44373#discussion_r434220679) [](ancestors = 434220679,434096192)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/44809](https://github.com/dotnet/roslyn/issues/44809) to follow up on adding DebugPlus to OptimizationLevel and making it public API. 

---
In reply to: [434257833](https://github.com/dotnet/roslyn/pull/44373#discussion_r434257833) [](ancestors = 434257833,434220679,434096192)